Jenkinsfile

```diff
@@ -16,60 +14,70 @@ pipeline {
     timestamps()
     timeout(time: 30, unit: 'MINUTES')
   }
-  environment {
-    PYTEST_PROCESSES = "${PYTEST_PROCESSES ?: "auto"}"
-    PYTEST_ADDOPTS =
-      "-n=${PYTEST_PROCESSES} " +
-      "--tb=short " +
-      "--color=yes " +
-      "--driver=SauceLabs " +
-      "--variables=capabilities.json"
-    PULSE = credentials('PULSE')
-    SAUCELABS = credentials('SAUCELABS')
-  }
   stages {
     stage('Lint') {
+      agent {
+        dockerfile true
+      }
       steps {
         sh "flake8"
       }
     }
     stage('Test') {
+      agent {
+        dockerfile true
+      }
+      environment {
+        PYTEST_PROCESSES = "${PYTEST_PROCESSES ?: "auto"}"
+        PYTEST_ADDOPTS =
+          "-n=${PYTEST_PROCESSES} " +
+          "--tb=short " +
+          "--color=yes " +
+          "--driver=SauceLabs " +
+          "--variables=capabilities.json"
```

Comment:
Rather than using `PYTEST_ADDOPTS` we can just put these arguments directly into the pytest command on line 43. This has the advantage of seeing the arguments in the console log.